app/src/main/java/com/ataulm/whatsnext/WhatsNextService.kt

```diff
@@ -85,3 +88,21 @@ private fun FilmRating.toApiValue(): String {
         FilmRating.FIVE -> "5"
     }
 }
+
+private class PopularFilmsThisWeekPagingSource(
+        private val letterboxdApi: LetterboxdApi,
+        private val filmSummaryConverter: FilmSummaryConverter
+) : PagingSource<String, FilmSummary>() {
+
+    override suspend fun load(params: LoadParams<String>): LoadResult<String, FilmSummary> {
+        val response = letterboxdApi.popularFilmsThisWeek(
+                cursor = params.key,
+                perPage = params.loadSize
+        )
+        return LoadResult.Page(
+                data = response.items.map { filmSummaryConverter.convert(it) },
+                prevKey = null, // API only allows paging forwards
```

Comment:
it's a shame it's not actually position based :( http://api-docs.letterboxd.com/#/definitions/FilmsResponse